server.go

```diff
@@ -262,11 +270,32 @@ func (s *Server) ReadSocket(packetPool *sync.Pool, reuseport bool) {
 
 // HTTPServe starts the HTTP server and listens perpetually until it encounters an unrecoverable error.
 func (s *Server) HTTPServe() {
+	var prf interface {
+		Stop()
+	}
+
+	once := sync.Once{}
+
+	if s.debug {
+		profileStartOnce.Do(func() {
+			prf = profile.Start()
+		})
+
+		defer func() {
```

Comment:
nitpick: we should probably explain that we're intentionally invoking prf.Stop twice (once in the deferral and once in the shutdown hook) just in case the shutdown hook doesn't get run for some reason

although now that I think about it, I'm not sure that case can ever actually happen... do we know if this deferral is necessary?